scvi/model/base/_pyromixin.py

```diff
@@ -0,0 +1,415 @@
+    """
+
+    def train(
+        self,
+        max_epochs: Optional[int] = None,
+        use_gpu: Optional[Union[str, int, bool]] = None,
+        train_size: float = 0.9,
+        validation_size: Optional[float] = None,
+        batch_size: int = 128,
+        early_stopping: bool = False,
+        plan_kwargs: Optional[dict] = None,
+        **trainer_kwargs,
+    ):
+        """
+        Train the model.
+
+        Parameters
+        ----------
+        max_epochs
+            Number of passes through the dataset. If `None`, defaults to
+            `np.min([round((20000 / n_cells) * 400), 400])`
+        use_gpu
+            Use default GPU if available (if None or True), or index of GPU to use (if int),
+            or name of GPU (if str, e.g., `'cuda:0'`), or use CPU (if False).
+        train_size
+            Size of training set in the range [0.0, 1.0].
+        validation_size
+            Size of the test set. If `None`, defaults to 1 - `train_size`. If
+            `train_size + validation_size < 1`, the remaining cells belong to a test set.
+        batch_size
+            Minibatch size to use during training.
+        early_stopping
+            Perform early stopping. Additional arguments can be passed in `**kwargs`.
+            See :class:`~scvi.train.Trainer` for further options.
+        plan_kwargs
+            Keyword args for :class:`~scvi.train.TrainingPlan`. Keyword arguments passed to
+            `train()` will overwrite values present in `plan_kwargs`, when appropriate.
+        **trainer_kwargs
+            Other keyword args for :class:`~scvi.train.Trainer`.
+        """
+        if max_epochs is None:
+            n_obs = self.adata.n_obs
+            max_epochs = np.min([round((20000 / n_obs) * 1000), 1000])
+
+        plan_kwargs = plan_kwargs if isinstance(plan_kwargs, dict) else dict()
+
+        if batch_size is None:
+            # use data splitter which moves data to GPU once
+            data_splitter = DeviceBackedDataSplitter(
+                self.adata,
+                train_size=train_size,
+                validation_size=validation_size,
+                batch_size=batch_size,
+                use_gpu=use_gpu,
+            )
+        else:
+            data_splitter = DataSplitter(
+                self.adata,
+                train_size=train_size,
+                validation_size=validation_size,
+                batch_size=batch_size,
+                use_gpu=use_gpu,
+            )
+        training_plan = PyroTrainingPlan(pyro_module=self.module, **plan_kwargs)
+
+        es = "early_stopping"
+        trainer_kwargs[es] = (
+            early_stopping if es not in trainer_kwargs.keys() else trainer_kwargs[es]
+        )
+        runner = TrainRunner(
+            self,
+            training_plan=training_plan,
+            data_splitter=data_splitter,
+            max_epochs=max_epochs,
+            use_gpu=use_gpu,
+            **trainer_kwargs,
```

Comment:
we should automatically add `early_stopping_monitor=train_loss_epoch` (or whatever it gets logged as) to the trainer kwargs, so the early stopping will work by default if user selects True.

Also, should we change the logged name to elbo?

Reply:
@adamgayoso where is validation loss logged?

Yes, I think would be good to rename that to "train_elbo_epoch"/"validation_elbo_epoch"

Reply:
well we don't actually do any validation loop with Pyro, I wasn't sure how to exactly in a universal way. Is it just `evaulate_loss`? And how does that work for regular black box VI? if there are no variational parameters for the val set?

Reply:
I think you can just save ELBO loss like this:
```python
validation_elbo = svi.evaluate_loss(*validation_args, **validation_kwargs)
```